applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp

```diff
@@ -656,61 +649,81 @@ void TrussElement3D2N::WriteTransformationCoordinates(
   KRATOS_CATCH("");
 }
 
+void TrussElement3D2N::AddExplicitContribution(
+    const VectorType& rRHSVector,
+    const Variable<VectorType>& rRHSVariable,
+    Variable<double >& rDestinationVariable,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
+
+    auto& r_geom = GetGeometry();
+
+    if (rDestinationVariable == NODAL_MASS) {
+        VectorType element_mass_vector(msLocalSize);
+        this->CalculateLumpedMassVector(element_mass_vector);
+
+        for (SizeType i = 0; i < msNumberOfNodes; ++i) {
+            double &r_nodal_mass = r_geom[i].GetValue(NODAL_MASS);
+            int index = i * msDimension;
+
+            #pragma omp atomic
+            r_nodal_mass += element_mass_vector(index);
+        }
+    }
+
+    KRATOS_CATCH("")
+}
+
 void TrussElement3D2N::AddExplicitContribution(
     const VectorType &rRHSVector, const Variable<VectorType> &rRHSVariable,
     Variable<array_1d<double, 3>> &rDestinationVariable,
-    const ProcessInfo &rCurrentProcessInfo) {
+    const ProcessInfo &rCurrentProcessInfo
+    )
+{
     KRATOS_TRY;
 
-    if (rRHSVariable == RESIDUAL_VECTOR &&
-        rDestinationVariable == FORCE_RESIDUAL) {
-
-    BoundedVector<double, msLocalSize> damping_residual_contribution =
-        ZeroVector(msLocalSize);
-      Vector current_nodal_velocities = ZeroVector(msLocalSize);
-      this->GetFirstDerivativesVector(current_nodal_velocities);
-      Matrix damping_matrix;
-      ProcessInfo temp_process_information; // cant pass const ProcessInfo
-      this->CalculateDampingMatrix(damping_matrix, temp_process_information);
-      // current residual contribution due to damping
-      noalias(damping_residual_contribution) =
-          prod(damping_matrix, current_nodal_velocities);
-
-
-      for (size_t i = 0; i < msNumberOfNodes; ++i) {
-        size_t index = msDimension * i;
-        array_1d<double, 3> &r_force_residual =
-            GetGeometry()[i].FastGetSolutionStepValue(FORCE_RESIDUAL);
-        for (size_t j = 0; j < msDimension; ++j) {
-  #pragma omp atomic
-          r_force_residual[j] +=
-              rRHSVector[index + j] - damping_residual_contribution[index + j];
+    if (rRHSVariable == RESIDUAL_VECTOR && rDestinationVariable == FORCE_RESIDUAL) {
+
+        BoundedVector<double, msLocalSize> damping_residual_contribution = ZeroVector(msLocalSize);
+        Vector current_nodal_velocities = ZeroVector(msLocalSize);
+        this->GetFirstDerivativesVector(current_nodal_velocities);
+        Matrix damping_matrix;
+        ProcessInfo temp_process_information; // cant pass const ProcessInfo
+        this->CalculateDampingMatrix(damping_matrix, temp_process_information);

```

Comment:
I think here you should use the lumped version you defined

Reply:
The one in the truss is already lumped (the mass terms)